### ISSUE TITLE
Difficult to override lexeme `ID`

### ISSUE THREAD

**Issue by SimonGreenhill** [lexibank/pylexibank#262]
I'm creating a dataset at the moment from a relational database and I'd like the lexeme IDs to be the primary keys from the database. I assumed that if I specified `ID=x` in the add_forms* call, then the ID would be kept, however, the code always calls `self.lexeme_id`:

https://github.com/lexibank/pylexibank/blob/8ae170cecb67f450b7a8cbaa56ded94281944b0f/src/pylexibank/cldf.py#L115-L120

Can we make lexeme_id be ignored if `ID` is given?. I see some lexibank datasets specify `ID` too which suggests other people expect this behavior (e.g. https://github.com/lexibank/lundgrenomagoa/blob/master/lexibank_lundgrenomagoa.py#L44).

**Comment by LinguList** [comment]
I remember we decided to leave Local_ID for that purpose of the original 
ID, as done in the dataset shared (this can also be in conflict with 
lexibank ID format, if I understand properly and is not checked). Would 
local_id be enough for you?

**Comment by LinguList** [comment]
I just checked here: 
https://github.com/intercontinental-dictionary-series/keypano/blob/main/lexibank_keypano.py

If ID is always created, we need indeed discuss that, as it means we 
would also have to revise datasets like the keypano, where we want to 
inherit IDS ids.

**Comment by SimonGreenhill** [comment]
Yes -- I'm using Local_ID but this feels like a workaround. If someone is explicitly specifying an ID then we should respect that.

At the very least silently replacing given ID's with a generated one is surprising and unexpected which could lead to glitches (I only spotted this when I was analysing the CLDF data sometime down the track and could not figure out why the list of lexemes I needed was wrong). It's worse because it's only the lexemes (add_form*) that do this and languages, parameters, etc are quite happy to have my manually curated IDs.

**Comment by xrotwang** [comment]
I think I agree that the call to `lexeme_id` should only happen in `add_forms_from_value`, because it's the potential splitting of values into several forms that made this functionality necessary in the first place. (Maybe `add_form_*` should have something like `kw.setdefault('ID', self.lexeme_id(...))` for backwards compat.)
